Userland/Libraries/LibCore/LocalSocket.cpp

```diff
@@ -55,25 +58,33 @@ LocalSocket::~LocalSocket()
 
 pid_t LocalSocket::peer_pid() const
 {
-#ifdef AK_OS_MACOS
+#if defined(__APPLE__)
     pid_t pid;
     socklen_t pid_size = sizeof(pid);
-
-    if (getsockopt(fd(), SOL_LOCAL, LOCAL_PEERPID, &pid, &pid_size) < 0) {
-        dbgln("LocalSocket: getsockopt failed, {}", strerror(errno));
-        VERIFY_NOT_REACHED();
-    }
-
-    return pid;
+#elif defined(__FreeBSD__)
+    struct xucred creds = {};
+    socklen_t creds_size = sizeof(creds);
 #else
     struct ucred creds = {};
     socklen_t creds_size = sizeof(creds);
+#endif
 
+#if defined(__APPLE__)
+    if (getsockopt(fd(), SOL_LOCAL, LOCAL_PEERPID, &pid, &pid_size) < 0) {
+#elif defined(__FreeBSD__)
+    if (getsockopt(fd(), SOL_LOCAL, LOCAL_PEERCRED, &creds, &creds_size) < 0) {
```

Comment:
Is this fix applicable to the other BSDs as well? It would be preferable to use some combination of the AK_OS_XXXX macros to the platform specific ones in that case. (e.g. defined(AK_OS_BSD_GENERIC) && !defined(AK_OS_MACOS) )

Reply:
OpenBSD seems to support `SO_PEERCRED` with a `struct` that's identical to Linux's `ucred`, but NetBSD supports LOCAL_PEEREID like macOS, so I will add it to the `ifdef` here.

Reply:
Yeah, the BSDs are pretty different in this area, so I don't think using `AK_OS_BSD_GENERIC` is possible here.

Reply:
That's unfortunate. If we get an implementation that works on each we can pretty up the os macros afterwards I suppose.